sql/core/src/main/scala/org/apache/spark/sql/jdbc/JDBCRDD.scala

```diff
@@ -331,7 +331,7 @@ private[sql] class JDBCRDD(
       case BooleanType => BooleanConversion
       case DateType => DateConversion
       case DecimalType.Unlimited => DecimalConversion(None)
-      case DecimalType.Fixed(d) => DecimalConversion(Some(d))
+      case DecimalType.Fixed(d, s) => DecimalConversion(Some((d, s)))
```

Comment:
This should be correct. Scala pattern extractors use tuples if they want to return multiple values.